airbyte-cdk/python/airbyte_cdk/sources/abstract_source.py

```diff
@@ -117,7 +117,7 @@ def read(
                         f"The stream '{configured_stream.stream.name}' in your connection configuration was not found in the source. "
                         f"Refresh the schema in your replication settings and remove this stream from future sync attempts."
                     )
-
+                    yield stream_status_as_airbyte_message(configured_stream.stream, AirbyteStreamStatus.INCOMPLETE)
```

Comment:
Based on the description `When trying to retrieve a stream, if the stream is missing from the source, the source should return a status of INCOMPLETE.`, it feels like the provided solution is not enough in the following cases:
* when we have two streams in the catalog that do not map to streams in the source. In that case, the first stream would yield a `AirbyteStreamStatus.INCOMPLETE` but not the second one
* when we  have `self.raise_exception_on_missing_stream == False`, the `AirbyteStreamStatus.INCOMPLETE` will not be emitted

Should we cover those two cases? The same concerns apply to the concurrent implementation.

Reply:
These are good questions.

> when we have two streams in the catalog that do not map to streams in the source. In that case, the first stream would yield an AirbyteStreamStatus.INCOMPLETE but not the second one

This depends on what we want to do with the behavior when an error is raised for non-existing streams. If we find that a stream is not in stream_instances, we raise an error and end the sync. As a result, we never check the next streams and force the user to change the configuration. 

> when we have self.raise_exception_on_missing_stream == False, the AirbyteStreamStatus.INCOMPLETE will not be emitted

This is fair. I have updated the code to emit the `INCOMPLETE` state regardless of whether `raise_exception_on_missing_stream` is true or false.

Reply:
> As a result, we never check the next streams and force the user to change the configuration.

This seems like a very bad UX, right? It does not seem like a good reason to me to stop syncing the other streams. I'm wondering if there is a reason for doing so...